geomesa-arrow/geomesa-arrow-jts/src/test/java/org/locationtech/geomesa/arrow/vector/GeometryVectorTest.java

```diff
@@ -542,6 +537,107 @@ public void testMultiPolygon() throws Exception {
     }
   }
 
+  @Test
+  public void testWKBGeometry() throws Exception {
+    WKTReader wktReader = new WKTReader();
+    WKTWriter wktWriter = new WKTWriter();
+
+    String geom1 = "POINT (0 20)";
+    String geom2 = "LINESTRING (30 10, 10 30, 40 45, 55 60, 56 60)";
+    String geom3 = "POLYGON ((35 10, 45 45, 15 40, 10 20, 35 10), (20 30, 35 35, 30 20, 20 30))";
+
+    try (RootAllocator allocator = new RootAllocator(Long.MAX_VALUE);
+         WKBGeometryVector doubles = new WKBGeometryVector("geoms", allocator, null)) {
+
+      Field doubleField = doubles.getVector().getField();
+
+      doubles.set(0, wktReader.read(geom1));
+      doubles.set(1, wktReader.read(geom2));
+      doubles.set(3, wktReader.read(geom3));
+      doubles.setValueCount(4);
+
+      Assert.assertEquals(4, doubles.getValueCount());
+      Assert.assertEquals(1, doubles.getNullCount());
+      Assert.assertEquals(geom1, wktWriter.write(doubles.get(0)));
+      Assert.assertEquals(geom2, wktWriter.write(doubles.get(1)));
+      Assert.assertEquals(geom3, wktWriter.write(doubles.get(3)));
+      Assert.assertNull(doubles.get(2));
+
+      // ensure field was created correctly up front
+      Assert.assertEquals(doubleField, doubles.getVector().getField());
+
+      // JNH:  I don't think there are fields?
+      //Assert.assertEquals(doubleField.getChildren(), PointVector.fields);  // FIX
```

Comment:
yeah there should just be the one field

Reply:
Cool.  I'll double-check that.  I think I saw an empty array the one thing I ran this in the debugger.

Reply:
Yeah, the field is just one field and no children.  I've updated the test.